src/ann_solo/reader.py

```diff
@@ -297,6 +497,78 @@ def read_mgf(filename: str) -> Iterator[MsmsSpectrum]:
         yield spectrum
 
 
+def read_query_file(filename: str) -> Iterator[MsmsSpectrum]:
+    """
+    Read all spectra from the given mgf, mzml, or mzxml file.
+
+    Parameters
+    ----------
+    filename: str
+        The mgf file name from which to read the spectra.
+
+    Returns
+    -------
+    Iterator[Spectrum]
+        An iterator of spectra in the given mgf file.
+    """
+    verify_extension(['.mgf', '.mzml', '.mzxml'],
+                     filename)
+
+    _, ext = os.path.splitext(os.path.basename(filename))
+
+    if ext == '.mgf':
+        return read_mgf(filename)
+    elif ext == '.mzml':
+        return read_mzml(filename)
+    elif ext == '.mzxml':
+        return read_mzxml(filename)
+
+##WIP: Support for compressed query files
+def read_query(filename: str) -> Iterator[MsmsSpectrum]:
+    """
+    Assess the query file format, whether compressed or not, and read its
+    content according to its extension.
+
+    Parameters
+    ----------
+    filename: str
+        The query file name from which to read the spectra.
+
+    Returns
+    -------
+    Iterator[Spectrum]
+        An iterator of spectra in the given mgf file.
+    """
+    verify_extension(['.mgf','.mzml','.mzxml','.zip','.gz','.xz'],
+                     filename)
+
+    _, ext = os.path.splitext(os.path.basename(filename))
+
+    if ext.lower() == '.gz':
+        with gzip.open(filename, 'rb') as gz_file:
+            gz_file_content = gz_file.read()
+            # create a file-like object from the content
+            file = io.StringIO(gz_file_content.decode())
```

Comment:
**question:** Does this cause memory copies? It might be pretty slow for large files.

Reply:
It does can be fixed by reading chunks with max buffer size or memory mapping. Facing issues with empty reads which need further investigation. To be done in the future.